packages/react-hooks/use-local-storage/useLocalStorage.ts

```diff
@@ -0,0 +1,48 @@
+import { Dispatch, SetStateAction, useCallback, useState } from "react";
+
+const storeValue = <T>(key: string, value: T): void => {
+  if (!window) return;
+
+  const valueSerialized = JSON.stringify(value);
+  localStorage.setItem(key, valueSerialized);
+};
+
+const getInitialStoredValue = <T>(key: string, initialValue: T): T => {
+  if (!window) return initialValue;
+
+  const storedValue = localStorage.getItem(key);
+  if (storedValue !== null) return JSON.parse(storedValue);
+  else {
+    storeValue(key, initialValue);
+    return initialValue;
```

Comment:
we can avoid having an `else` statement when the `if` has a return

Reply:
also, the read and JSON.parse can fail but we need to make it possible to recover from that behaviour. We can change this to something like

```ts
const getStoredValue = <T>(key: string, initialValue: T): T => {
    // Prevent build error "window is undefined" but keep keep working
    if (typeof window === "undefined") {
      return initialValue;
    }

    try {
        const item = window.localStorage.getItem(key);
        return item ? JSON.parse(item) : initialValue;
    } catch (err) {
        console.log(err);
        return initialValue;
    }
}
```